[GitlabIssueBridge] Add bridge

Mynacol opened the pull request:
This bridge provides content for GitLab issues and merge requests, similar to the GithubIssueBridge.

Runs smoothly for ~two weeks for me.

Mynacol committed:
[GitlabIssueBridge] new bridge

This tracks issue comments on arbitrary gitlab projects.

Mynacol committed:
[GitlabIssueBridge] Prepare for Merge Request support + fixes

- Proper UIDs
- Default bridge name fixed
- Fix cache identifiers
- Add TODOs

Mynacol committed:
[GitlabIssueBridge] creation timestamp preferred

And prefer original author over editor.

Mynacol committed:
[GitlabIssueBridge] Do not add date to item title

Prettier without it.

Mynacol committed:
[GitlabIssueBridge] Support Merge Requests

This bridge can now generate feeds for Merge Requests.

Mynacol committed:
[GitlabIssueBridge] typo

Mynacol committed:
[GitlabIssueBridge] Fix Img src attr in comments

Mynacol committed:
[GitlabIssueBridge] Fix function call

Mynacol committed:
[GitlabIssueBridge] Fix test

Use gitlab.com if no h parameter was given.
Fixes a phpunit test.

Mynacol committed:
[GitlabIssueBridge] linting

Mynacol committed:
[GitlabIssueBridge] Add MR support to description

dvikan commented:
gitlab has really good feeds so I assume they don't include issue and merge comments in their feed?

Move the `collectData` method to the top. Inline those class constants that are used to build urls. Why do manual caching inside the bridge when the entire bridge output is cached?

Mynacol committed:
[GitlabIssueBridge] Move function collectData

Mynacol committed:
[GitlabIssueBridge] rm single-use class constants

Mynacol committed:
[GitlabIssueBridge] Remove manual caching

Just depend on rss-bridges built-in caching.

Mynacol commented:
> gitlab has really good feeds so I assume they don't include issue and merge comments in their feed?

Exactly. They just don't have feeds for individual issues/MRs.

The issues should be resolved (see the last three commits).
I moved `collectData` to a sensible place IMO.
I removed the manual caching, as it is not needed here.

I additionally rebased everything on master.

dvikan commented:
The `collectData` method should be at the top because it is usually the first method you want to read. The `parseMRDescription()` should be renamed to `parseMergeRequestDescription`. `getProjectURI()` should be inlined. I did some testing with `gitlab.gnome.org` and `GNOME/gnome-shell`. It's good. Thanks.
